lib/bundler/shared_helpers.rb

```diff
@@ -305,7 +305,7 @@ def set_bundle_variables
       exe_file = File.expand_path("../../../exe/bundle", __FILE__)
 
       # for Ruby core repository testing
-      exe_file = File.expand_path("../../../bin/bundle", __FILE__) unless File.exist?(exe_file)
+      exe_file = File.expand_path("../../../libexec/bundle", __FILE__) unless File.exist?(exe_file)
```

Comment:
I think there was a bug here, although this code is only for tests and the unit test was also checking for the wrong file, so it doesn't really make a difference. This is supposed to use the actual bundle bin, not a bundler binstub.